### ISSUE TITLE
Restore removed self-assigned articles to the pool of available articles 

### ISSUE THREAD

**Issue by Ibom99** [WikiEducationFoundation/WikiEduDashboard#5549]
### Current Behavior:
At the moment, the programs and event dashboard doesn't move self-assigned articles back to the pool of available articles when remove is selected, it deletes the article completely from the program.

### Desired Behavior:
The self-assigned articles when removed should return to the list of available articles it should not be deleted from the program entirely. 

### Additional context:
1. Self-Assigned Article

![Screenshot (70)](https://github.com/WikiEducationFoundation/WikiEduDashboard/assets/104465078/a1b9a0f9-7095-40e0-909b-0df03d2f894d)

2. After selecting remove 

![Screenshot (71)](https://github.com/WikiEducationFoundation/WikiEduDashboard/assets/104465078/17cf54b1-2926-43c7-94aa-33cbe16ba03f)

3. Article not in the list of the available articles
![Screenshot (72)](https://github.com/WikiEducationFoundation/WikiEduDashboard/assets/104465078/c16e82e4-caf9-4d83-8d6f-6320cea04aee)

**Comment by ragesoss** [comment]
This feature would touch a lot of files, mostly involving the `Assignment` model, along with related controllers and front-end components.

**Comment by fillingtothemomo** [comment]
@ragesoss can I take this issue up?

**Comment by ragesoss** [comment]
@fillingtothemomo yes, but I think this is a fairly complicated issue that, and it may take a number of iterations to figure out exactly what the implementation should look like.

**Comment by fillingtothemomo** [comment]
@ragesoss Can you suggest some newcomer friendly issues for me to work on?

**Comment by ragesoss** [comment]
@fillingtothemomo I think you should give #5489 a try. It looks like the other person isn't active on it anymore.

**Comment by ragesoss** [comment]
A few clarifications of this issue:

1. This should account for the possibility that the removed article is already present as an Available Article (and should simply remove the article, in that case).
2. This new behavior should only apply to courses that are using Available Articles in the first place, but we don't actually have a good way to detect that option right now. Basically, we want this new behavior only for a course where the instructor chose the 'Instructor prepares a list' option in the assignment wizard. I think a good way to address this would be to add `choose_articles_from_list` as a tag in the `researchwrite/wizard.yml` file so that the courses where this should apply will have the `choose_articles_from_list` Tag. Then the behavior can be controlled on the backend in the controller by determining whether the course has that tag.

**Comment by shishiro26** [comment]
@ragesoss I’d like to work on this. What’s the current status of the issue? While trying to recreate it in my dev environment, I noticed that both self-assigned articles and admin-assigned articles are being removed from the available articles. Is this the issue that needs fixing, or should the focus be only on self-assigned articles?

**Comment by ragesoss** [comment]
@shishiro26: I think the best behavior would be that, if an article gets assigned via the Available Articles list (as opposed to assigned by just entering a title), then unassigning it should restore it to the Available Articles list.
